bin/etc-update

```diff
@@ -33,12 +33,13 @@ get_config() {
 }
 
 OS_RELEASE_ID=$(source /etc/os-release >/dev/null 2>&1; echo "${ID}")
+OS_RELEASE_ID_LIKE=$(source /etc/os-release >/dev/null 2>&1; echo "${ID_LIKE}")
```

Comment:
You can extract multiple variables from a sourced file in this way:
```
eval "$(source /etc/os-release &>/dev/null; declare -A OS_RELEASE_DATA=([ID]="${ID}" [ID_LIKE]="${ID_LIKE}"); declare -p OS_RELEASE_DATA)"
```
And then use `"${OS_RELEASE_DATA[ID]}"` and `"${OS_RELEASE_DATA[ID_LIKE]}"`.

So either directly `case ":${OS_RELEASE_DATA[ID]}:${OS_RELEASE_DATA[ID_LIKE]}:" in` or assign `OS_RELEASE_ID="${OS_RELEASE_DATA[ID]}"` and `OS_RELEASE_ID_LIKE="${OS_RELEASE_DATA[ID_LIKE]}"` and use `case ":${OS_RELEASE_ID}:${OS_RELEASE_ID_LIKE}:" in`.

Reply:
Don't abuse `eval`. Ever.

Reply:
My solution avoids spawning 2 subprocesses and opening `/etc/os-release` twice for reading.

It is also possible to send contents of variables to temporary files, but that only avoids subprocess problem.

So there is no consensus that this would be any abuse of `eval`.

Reply:
There are many solutions to every problem, and the one using `eval` is almost always wrong.

Reply:
Here's a way to read the file once, without using eval:
```bash
for var in OS_RELEASE_ID OS_RELEASE_ID_LIKE; do
	read -r -d '' || break
	declare ${var}="${REPLY}"
done < <(source /etc/os-release >/dev/null 2>&1; printf -- '%s\0%s\0' "${ID}" "${ID_LIKE}")
```

Reply:
You're all sort of missing the forest for the trees. Not only is eval (a dangerous thing which should only be used with care, and this doesn't qualify) completely unnecessary, you don't need two variables to begin with.

```
OS_RELEASE_POSSIBLE_IDS=$(source /etc/os-release >/dev/null 2>&1; echo "${ID}:${ID_LIKE}")

case ":${OS_RELEASE_POSSIBLE_IDS}:" in
```

The variables are used once, right here and now. They don't need to be uniquely available. No read loop hacks, no frivolous use of eval, no opening the file twice for reading, no forking bash into a second subshell on top of the first.